elements/lisk-p2p/src/utils/validate.ts

```diff
@@ -174,6 +175,10 @@ export const validatePeerInfoSchema = (rawPeerInfo: unknown): P2PPeerInfo => {
 			height: height && isNumeric(height.toString()) ? +height : 0,
 			...restOfProtocolPeer,
 		},
+		internalState: {
+			advertiseAddress:
+				advertiseAddress === false ? (advertiseAddress as boolean) : true,
```

Comment:
forgot to mention that this `peerInfo` might have `advertiseAddress` as a string since its coming from the network
we can check `advertiseAddress == false ? false: true`